client/normal_value_test.go

```diff
@@ -1622,3 +1622,28 @@ func TestNormalValue_ToArrayOfNormalValues(t *testing.T) {
 		})
 	}
 }
+
+// This test documents a bug where array values
+// were not returning the correct value for IsNillable
+// and were also not convertible to a normal nil kind.
+func TestArrayValue_IsNillable(t *testing.T) {
+	fieldKinds := []FieldKind{
+		FieldKind_BOOL_ARRAY,
+		FieldKind_INT_ARRAY,
+		FieldKind_FLOAT_ARRAY,
+		FieldKind_STRING_ARRAY,
+		FieldKind_NILLABLE_BOOL_ARRAY,
+		FieldKind_NILLABLE_INT_ARRAY,
+		FieldKind_NILLABLE_FLOAT_ARRAY,
+		FieldKind_NILLABLE_STRING_ARRAY,
```

Comment:
todo: sorry, but I'm trying to figure out what this PR actually tries to achieve.
Looking at this test I'm a bit confused.
We have the following array types:
1. FieldKind_<type>_ARRAY - NOT nillable array with NOT nillable elements
2. FieldKind_NILLABLE_<type>_ARRAY - NOT nillable array with nillable elements
3. FieldKind_<type>_NILLABLE_ARRAY - nillable array with NOT nillable elements
4. FieldKind_NILLABLE_<type>_NILLABLE_ARRAY - nillable array with nillable elements
At least this is how these different array kinds are supposed to be distinguished.

At the moment `IsNillable` on `ScalarArrayKind` return true IIRC only elements are nillable, but it's no ideal. I also planned to put some effort on this.

So please clarify which of those 4 types you are changing and how.

Reply:
There was a discord conversation around this, all the scalar arrays are nillable atm, the enum values are just stuck with the old names.  e.g. `FieldKind__ARRAY` is nillable, with non-nillable elements.

Reply:
At the moment, `FieldKind__ARRAY` is a Nillable array with nillable elements. That is where we all get confused. This needs to be fix. This PR ensures that things work as they should given the current naming.

Reply:
`FieldKind__NILLABLE_ARRAY` and `FieldKind_NILLABLE__NILLABLE_ARRAY` don't exist at the moment.

Reply:
looks like I missed this discord conversation. All right then